resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/BaseDriverConfigurationStep.scala

```diff
@@ -133,13 +144,17 @@ private[spark] class BaseDriverConfigurationStep(
         .withNodeSelector(nodeSelector.asJava)
         .endSpec()
       .build()
+
+    val (resolvedDriverPod, resolvedDriverContainer) =
+      mountSecretsBootstrap.mountSecrets(baseDriverPod, driverContainer)
```

Comment:
Don't do the bootstrap in the base step. Create a separate step for this, and switch on whether we include the step (as opposed to whether or not we include the bootstrap) in the orchestrator.

Reply:
Done.